ocp_resources/user_defined_network.py

```diff
@@ -68,6 +68,74 @@ def to_dict(self) -> None:
 
     # End of generated code
 
+    @classmethod
+    def is_ready_condition(cls, condition: dict) -> bool:
+        """
+        Check if the given condition indicates that the UserDefinedNetwork is ready.
+
+        Args:
+            condition (dict): A dictionary representing the condition of the UserDefinedNetwork.
+
+        Returns:
+            bool: True if the condition indicates the UserDefinedNetwork is ready, False otherwise.
+        """
+        return (
+            condition.get("reason") == cls.Condition.Reason.NETWORK_ATTACHMENT_DEFINITION_READY
+            and condition.get("status") == cls.Condition.Status.TRUE
+            and condition.get("type") == cls.Condition.Type.NETWORK_READY
+        )
+
+    @classmethod
+    def is_sync_error_condition(cls, condition: dict) -> bool:
+        """
+        Check if the given condition indicates a synchronization error for the UserDefinedNetwork.
+
+        Args:
+            condition (dict): A dictionary representing the condition of the UserDefinedNetwork.
+
+        Returns:
+            bool: True if the condition indicates a synchronization error, False otherwise.
+        """
+        return (
+            condition.get("reason") == cls.Condition.Reason.SYNC_ERROR
+            and condition.get("status") == cls.Condition.Status.FALSE
```

Comment:
If the status is `false`, doesn't it mean that the error condition (SYNC_ERROR) you are checking here is not filled?

Reply:
It's not mean the condition is not filled, it means that the condition is not applicable, which is correct as the condition represent sync error, a negative condition.
See k8s documentaion for more info:
https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions

Reply:
Oh, my bad. I confused the condition's `reason` with its type. I now see that `SYNC_ERROR` is the `reason`, while the `type` is `NETWORK_READY`, which is false in this check.
Sorry for missing that.